api/api.go

```diff
@@ -806,3 +814,35 @@ func (b *BlockChainAPI) prepareBlockResponse(
 
 	return blockResponse, nil
 }
+
+func (b *BlockChainAPI) getCadenceHeight(
+	blockNumberOrHash *rpc.BlockNumberOrHash,
+) (uint64, error) {
+	if number, ok := blockNumberOrHash.Number(); ok {
+		height := uint64(0)
+		if number >= 0 {
```

Comment:
we should handle special values as well:
```
	SafeBlockNumber      = BlockNumber(-4)
	FinalizedBlockNumber = BlockNumber(-3)
	LatestBlockNumber    = BlockNumber(-2)
	PendingBlockNumber   = BlockNumber(-1)
```
for now all can equal to latest block since we don't have this special states for EVM

Reply:
We do handle that actually, if `number < 0`, we use `ExecuteScriptAtLatestBlock`, instead of `ExecuteScriptAtBlockHeight`.

Reply:
Here you can see the test case which asserts this behavior:
https://github.com/onflow/flow-evm-gateway/pull/192/files#diff-7bfc48a9431bb23a8334210b28a437344f6ceee9a523d2484ca63ec2f3c1a8a9R54-R59
```javascript
// get balance at special block tags
let blockTags = ["latest", "pending", "safe", "finalized"]
for (let blockTag of blockTags) {
    receiverWei = await web3.eth.getBalance(receiver.address, blockTag)
    assert.equal(receiverWei, transferValue)
}
```